site/content/en/docs/manual/basics/attach-cloud-storage.md

```diff
@@ -166,9 +166,68 @@ Prepare the dataset as in the point [prepare dataset](#prepare-dataset).
 
 Now you can attach new cloud storage into CVAT.
 
+## Using Google cloud storage
+
+### Create Google account
+
+First, create a Google account, go to [account login page](https://accounts.google.com/) and click `Create account`.
+After [registering in 3 steps](https://console.cloud.google.com/freetrial) enter the required data
+and accept the terms of service (you'll need credit card information to register).
+
+### Create a bucket
+
+Your first project will be created automatically, you can see it on the [cloud resource manager page](https://console.cloud.google.com/cloud-resource-manager).
+To create a bucket, go to the [cloud storage page](https://console.cloud.google.com/storage/browser)
+and press `Create bucket`. Next, enter the name of the bucket, add labels if necessary, select the type of location
+for example region and the location nearest to you, select storage class, when selecting access control
+you can enable `Enforce public access prevention on this bucket` (if you plan to have anonymous access to your bucket,
+it should be disabled) you can select `Uniform` or `Fine-grained` access control, if you need protection of your object data
+select protect object data type. When all the information is entered click `Create` to create the bucket.
+
+![](/images/google_cloud_storage_tutorial1.jpg)
+
+### Upload
+
+Prepare the dataset as in the point [prepare dataset](#prepare-dataset).
+
+To upload files, you can simply drag and drop files and folders into a browser window
+or use the `upload folder` and/or `upload files`.
+
+### Access permissions
+
+To access Google Cloud Storage from CVAT you will need a `Project ID`
+you can find it by going to [cloud resource manager page](https://console.cloud.google.com/cloud-resource-manager)
+
+![](/images/google_cloud_storage_tutorial5.jpg)
+
+#### Create a service account and key file
+
+To access your bucket you need a key file and a service account. To create a service account,
+go to `IAM & Admin`/`Service Accounts` and press `Create Service Account`. Enter your account
+name and click `Create And Continue`. Select a role for example `Basic`/`Viewer`.
```

Comment:
I'm not sure about this role. Why isn't it `Cloud Storage/....` ?

Reply:
Role from `Cloud Storage`/ `…` not working, and role `Storage Legacy` missing for service account
Working role:
Current `Basic`/`Viewer`
`Dataflow` / `Dataflow Viewer`
`Dateproc` / `Dateproc Viewer`